### ISSUE TITLE
disablePast doesnt disable AM for selection

### ISSUE THREAD

**Issue by ng9000** [mui/mui-x#12159]
### Search keywords

disablePast

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Disable Past should disable the selection of AM in StaticDatePicker but it doesn't so you can get something in the output like {today's date 5:00 AM} which will be wrong for most scenarios. Using this [date picker](https://mui.com/x/react-date-pickers/date-time-picker/#landscape-orientation)

### Current behavior

AM is not disabled if disablePast is passed.

### Expected behavior

AM should be disabled when passing disablePast.

### Context

I am trying to get date and time to schedule a call which has to be some future time and not past.

### Your environment

```
import {
  LocalizationProvider,
  StaticDateTimePicker,
} from "@mui/x-date-pickers";
import dayjs from "dayjs";


<LocalizationProvider dateAdapter={AdapterDayjs}>
      <StaticDateTimePicker
          orientation="landscape"
          onChange={(x) => handleChange(x)}
          ampm
          disablePast
          defaultValue={dayjs(new Date())}
          slotProps={{ actionBar: { actions: [] } }}
          className="mt-4"
          sx={{
           color: "success.main",
            "& .MuiDateTimePickerToolbar-ampmLabel": {
             fontWeight: "900",
          },
         }}
     />
</LocalizationProvider>
```

**Comment by LukasTy** [comment]
This change needs to be applied (or checked) on the various Toolbar components including time (`DateTimeToolbar`, `TimeToolbar`, `DateTimeRangeToolbar`), the `TimeClock` as well as `MultiSectionDigitalClock` components.

**Comment by A-s-h-o-k** [comment]
@LukasTy I noticed the issue with #12159. I believe I can find a solution and would like to take ownership of this task. Please let me know if you’re okay with me handling it.

**Comment by LukasTy** [comment]
> I noticed the issue with #12159. I believe I can find a solution and would like to take ownership of this task. Please let me know if you’re okay with me handling it.

I appreciate your interest! 🙏 
I have assigned the issue to you. 😉 
If you have any questions, feel free to ask. 👍

**Comment by A-s-h-o-k** [comment]
@LukasTy regarding #12159 

AM/PM should also disable on MobileTimePicker component for disableFeature or disablePast props

![image](https://github.com/user-attachments/assets/713a1812-4322-4206-af98-34015c6a1941)

**Comment by LukasTy** [comment]
> AM/PM should also disable on MobileTimePicker component for disableFeature or disablePast props

@A-s-h-o-k Great observation. 👍 
It should disable the toolbar buttons as well as buttons in the clock when the `ampmInClock` prop is present.

**Comment by A-s-h-o-k** [comment]
@LukasTy  Got it

**Comment by A-s-h-o-k** [comment]
Hi @LukasTy ,
Could you please review the PR I submitted for **DesktopTimePicker**?
I’ve also made progress on **MobileTimePicker** and plan to raise a separate PR for it once the work is complete.

Thank you!
